sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuPartitioning.scala

```diff
@@ -241,4 +252,19 @@ trait GpuPartitioning extends Partitioning {
       }
     }
   }
+
+  private var memCopyTime: Option[GpuMetric] = None
+
+  /**
+   * Setup sub-metrics for the performance debugging of GpuPartition. This method is expected to
+   * be called at the query planning stage. Therefore, this method is NOT thread safe.
+   */
+  def setupDebugMetrics(metrics: Map[String, GpuMetric]): Unit = {
+    // Check and set GpuPartitioning.CopyToHostTime
+    if (memCopyTime.isEmpty) {
+      metrics.get(GpuPartitioning.CopyToHostTime).foreach { metric =>
+        memCopyTime = Some(metric)
+      }
+    }
```

Comment:
I don't think we should ignore a new metric being specified even if we have an old one.  Shouldn't happen in practice, but if it does, we probably want the later one, not the earlier one.

Also the design gets simpler if we leverage NoopMetric to eliminate the checks when using it later.
```suggestion
    memCopyTime = metrics.get(GpuPartitioning.CopyToHostTime).getOrElse(NoopMetric)
```